infra/staging/main.tf

```diff
@@ -151,6 +151,66 @@ resource "google_storage_bucket_iam_member" "frontend_bucket_viewer" {
   role   = "roles/storage.objectViewer"
   member = "allUsers"
 }
+#######################################
+## Load balancer for frontend bucket ##
+#######################################
+
+// Following the guide here: https://cloud.google.com/cdn/docs/setting-up-cdn-with-bucket
+
+// Enable Compute Engine API to setup an IP address
+resource "google_project_service" "compute_engine_api" {
+  project = data.google_project.project.project_id
+  service = "compute.googleapis.com"
+}
+
+// Setup IP address for load balancer
+resource "google_compute_global_address" "call_home_frontend_ip" {
+  project = data.google_project.project.project_id
+  name = "call-home-frontend-ip"
+}
+
+// Backend bucket with CDN policy with default ttl settings
+resource "google_compute_backend_bucket" "call_home_frontend_lb" {
+  name = "call-home-frontend-lb"
+  description = "Load balancer for the frontend GCS bucket"
+  project = data.google_project.project.project_id
+  bucket_name = google_storage_bucket.call_home_frontend.name
+  enable_cdn = true
+
+  cdn_policy {
+    cache_mode = "CACHE_ALL_STATIC"
+    client_ttl = 3600
+    default_ttl = 3600
+    max_ttl = 86400
+    negative_caching = true
+    serve_while_stale = 86400
+  }
+}
+
+// URL map for the backend bucket
+resource "google_compute_url_map" "call_home_frontend_lb_url_map" {
+  name = "call-home-frontend-lb-url-map"
```

Comment:
Confusingly, this maps to two places  in the GCP console:

- Load balancer: https://console.cloud.google.com/net-services/loadbalancing/details/http/call-home-frontend-lb-url-map?project=call-home-staging
- URL map field in https://console.cloud.google.com/net-services/loadbalancing/advanced/targetHttpProxies/details/call-home-frontend-lb-proxy?project=call-home-staging

Keeping the `url-map` name is okay, though it is a bit weird that the Load Balancer is named `*-url-map`. Maybe we should rename this to `call-home-frontend-lb` or `call-home-lb` (if we think we will reuse it for the Strapi App Engine app).

If we rename it, take note that this sort of networking resource is a bit finicky to delete in Terraform because networking requires a lot of external interactions and resource creation/teardown take a long time. Fortunately, we can change the internal Terraform address using `terraform state mv` (https://stackoverflow.com/questions/70418172/terraform-rename-a-resource-without-deleting) and I _think_ we can rename the `.name` property. (For this reason, it doesn't matter if we name it either `call-home-frontend-lb` or `call-home-lb`, since we can rename it afterwards.)

Reply:
Okay, I think I prefer to drop the `_url_map` part (though I won't insist), since:

- If we're viewing the url map, it's redundant. (see also https://github.com/bettersg/call-home/pull/145#discussion_r1139707213)
- If we're viewing the load balancer, it's misleading.